v3/src/components/map/components/map-interior.tsx

```diff
@@ -9,20 +9,31 @@ import {MapPointLayer} from "./map-point-layer"
 import {isMapPolygonLayerModel} from "../models/map-polygon-layer-model"
 import {MapPolygonLayer} from "./map-polygon-layer"
 import { DataConfigurationContext } from "../../data-display/hooks/use-data-configuration-context"
+import { useTileModelContext } from "../../../hooks/use-tile-model-context"
 
 interface IProps {
   pixiPointsArrayRef:  React.MutableRefObject<PixiPoints[]>
 }
 
 export const MapInterior = observer(function MapInterior({pixiPointsArrayRef}: IProps) {
   const mapModel = useMapModelContext()
+  const { transitionComplete: tileTransitionComplete } = useTileModelContext()
 
   useMapModel()
 
   const onSetPixiPointsForLayer = useCallback((pixiPoints: PixiPoints, layerIndex: number) => {
     pixiPointsArrayRef.current[layerIndex] = pixiPoints
   }, [pixiPointsArrayRef])
 
+
+  // Ensure the map rescales when its tile's CSS transition is complete to compensate for any changes to the tile's
+  // size that may have occurred after the map was initially rendered.
+  useEffect(function rescaleOnTileTransitionEnd() {
+    if (tileTransitionComplete) {
+      mapModel.rescale()
+    }
+  }, [tileTransitionComplete, mapModel])
+
```

Comment:
I'm not sure this is the best way to do this since it will result in two calls to `rescale` on the map model, the other being in `useMapModel`. I'm not sure there's a way around that, though. Unless it would be OK to remove the call to `rescale` in `useMapModel`? It's not clear to me whether that may be required for other things.

Reply:
I removed the call in useMapModel as this call handles all the relevant situations.